app/code/Magento/Deploy/Package/PackagePool.php

```diff
@@ -241,6 +270,47 @@ private function ensureRequiredLocales(array $options)
         }
     }
 
+    /**
+     * Get used store and admin user locales.
+     *
+     * @return array
+     * @throws InvalidArgumentException if unknown locale is provided by the store configuration
+     */
+    private function getUsedLocales()
+    {
+        $usedLocales = array_merge(
+            $this->storeView->retrieveLocales(),
+            $this->getAdminUserInterfaceLocales()
+        );
+
+        return array_map(
+            function ($locale) {
+                if (!$this->locale->isValid($locale)) {
+                    throw new InvalidArgumentException(
+                        $locale . ' argument has invalid value, run info:language:list for list of available locales'
+                    );
+                }
+
+                return $locale;
+            },
+            array_unique($usedLocales)
+        );
+    }
+
+    /**
+     * Get admin user locales
+     *
+     * @return array
+     */
+    private function getAdminUserInterfaceLocales()
+    {
+        $locales = [AppInterface::DISTRO_LOCALE_CODE];
+        foreach ($this->userCollection as $user) {
```

Comment:
Right now static content deploy could be ran on build environment that not contains any DB, details here https://devdocs.magento.com/guides/v2.4/config-guide/deployment/ .
Looks like this change brings back dependency to DB and breaks this ability. Could you double check it?
Rather than that I would suggest getting list of locales from config.php.
I'm not sure if we already could define list of locales used in admin in `config.php` file, but even if it doesn't - I think we could add such option.

Reply:
The dependency on the DB has been removed by adding the following code:

```php
$connections = $this->deploymentConfig->get(ConfigOptionsListConstants::CONFIG_PATH_DB_CONNECTIONS, []);
if (empty($connections)) {
    return $this->usedAdminLocales;
}
```

Admin locales can now be added using the following `app/etc/env.php` config:
```php
    'force_deploy_admin_locales' => [
        'en_GB',
        'de_DE',
    ]
```